src/implementations/environments.jl

```diff
@@ -147,6 +148,30 @@ end
 get_reward(env::RewardOverriddenEnv, args...; kwargs...) =
     foldl(|>, env.processors; init = get_reward(env.env, args...; kwargs...))
 
+
+#####
+# MaxTimeoutEnv
+#####
+
+mutable struct MaxTimeoutEnv{E<:AbstractEnv} <: AbstractEnv
+    env::E
+    max_t::Int
+    current_t::Int
+end
+
```

Comment:
Add a constructor `MaxTimeoutEnv(max_t)` to allow chaining multiple wrappers?

Reply:
Sure. So something like this?
```
# partial constructor to allow chaining
MaxTimeoutEnv(max_t, current_t) = env -> MaxTimeoutEnv(env, max_t, current_t)
```
How about keeping both `max_t` and `current_t` in the partial constructor?

Reply:
You can set `current_t ` as an optional keyword argument.

Reply:
Ok